src/ekat/io/ekat_yaml.cpp

```diff
@@ -303,47 +304,88 @@ std::string write_param<char> (const char& t) {
   return (t ? "true" : "false");
 }
 
-void write_parameter_list (const ParameterList& params, std::ostream& out, int indent) {
```

Comment:
Using a `YAML::Emitter`, we don't have to keep track of indentation and other shenanigans. It's the right thing to do: let the library take care of the YAML formatting.